### PR TITLE
Gen 116

### PR DESCRIPTION
**Pull request by fribas84**
added referral code and linters

### CODE REVIEW

**Comment by sunnyalgo** [comment]
That's great, and I would like to discuss about the referralCode and diburseBounty more..

File: contracts/Recruitment.sol [hunk 1]
```diff
@@ -51,7 +48,11 @@ contract Recruitment is Ownable, ReentrancyGuard {
     mapping(uint256 => FrontDoorStructs.Candidate[]) public candidateListForJob; // list of candidates for a job
     mapping(uint256 => FrontDoorStructs.Candidate) public jobCandidatehire;
 
-    mapping(address=>uint256) public bountyClaim;
+    mapping(address => uint256) public bountyClaim;
+
+    mapping(uint16 => bytes32) public JobIdtoTeferralCodeList;
```
Comment (codingnuclei) [inline contracts/Recruitment.sol:53]: is this unused?

File: contracts/Recruitment.sol [hunk 2]
```diff
@@ -196,33 +197,38 @@ contract Recruitment is Ownable, ReentrancyGuard {
      * @notice Registers a referral
      * @param jobId The job ID already registered with the contract.
      * @param refereeMail The email of the referee.
+     * @param referralCode The referral code of the referee, it must be unique and hashed.
      */
     function registerReferral(
         uint256 jobId,
-        bytes32  refereeMail
+        bytes32 refereeMail,
+        bytes32 referralCode
```
Comment (codingnuclei) [inline contracts/Recruitment.sol:205]: Ideally the referralCode is generated SC side?

**Review by codingnuclei — CHANGES_REQUESTED** [review]
Some small observations :)

File: contracts/Recruitment.sol (hunk 2 above)
Comment (sunnyalgo) [reply contracts/Recruitment.sol:205]: That's planned to get the referral code from front right now. like Jsontoken.

File: contracts/Recruitment.sol (hunk 1 above)
Comment (sunnyalgo) [reply contracts/Recruitment.sol:53]: That's used.

**Pull request closed without merging**